### PR TITLE
[NT-2163] Removed Deprecated Comments Feature and Related Feature Flags

### PR DESCRIPTION
**Pull request by msadoon**
# 📲 What

Part of the clean up after having added the new GQL based comments features. 

# 🤔 Why

Its good to do some clean up of the pre-existing feature now that we know the new feature is stable and a successful replacement.

The `OptimizelyFeature.commentFlaggingEnabled` is still in place because we haven't built the feature yet.

# 🛠 How

Started removing these cases from the `OptimizelyFeature`
- commentThreading
- commentThreadingRepliesEnabled

Then removed every file that had `Deprecated` prefixed to it. <-- _This was really helpful, very glad we did this first, otherwise finding every connection would have taken a much longer time._
- `DeprecatedCommentsDataSource`
- `DeprecatedCommentsDataSourceTests`
- `DeprecatedCommentCell`
- `DeprecatedCommentsEmptyStateCell`
- `DeprecatedCommentsViewController`
- `DeprecatedCommentsViewControllerTests`
- `DeprecatedComments.storyboard`
- `DeprecatedCommentsEnvelope`
- `DeprecatedCommentsEnvelopeTemplates`
- `DeprecatedCommentStyles`
- `DeprecatedCommentCellViewModel`
- `DeprecatedCommentCellViewModelTests`
- `DeprecatedCommentsEmptyStateCellViewModel`
- `DeprecatedCommentsEmptyStateCellViewModelTests`
- `DeprecatedCommentsViewModel`
- `DeprecatedCommentsViewModelTests`
- `DeprecatedCommentLenses`
- `DeprecatedCommentTemplates`
- `DeprecatedCommentDialogViewController`
- `DeprecatedCommentDialogViewModel`
- `DeprecatedAuthor` -> renamed to `ActivityComment`
- `DeprecatedComment` -> renamed to `ActivityCommentAuthor`

Since `DeprecatedAuthor` and `DeprecatedComment` are tied to `GET /v1/projects/\(project.id)/activities` and there is no GraphQL entity representing a **Project Activity**, its not possible at this time to remove the models entirely. The `Comment` model relies on the GQL properties to map to the `v1` properties, which is not entirely within the scope of this PR, and also would require a lot of overhead to complete, given the scope of `ProjectActivitiesViewController` and its many use cases. Plus I think this requires a larger product discussion.

It was possible however to remove
  - `POST /v1/projects/\(u.projectId)/updates/\(u.id)/comments`
  - `POST /v1/projects/\(p.id)/comments`
 from the `Service` and use the newly built `postComment(input: PostCommentInput)` to update the `CommentsDialogViewController` so we can get that for free which is nice.
Small note that we should maybe consider looking at this flow again, as replies are not treated the same way as replies in our new comment threading feature. Also the old comment dialog is being used for posting comments. Much, much larger discussion imo.

# ✅ Acceptance criteria

- [ ] Ensure no regressions were introduced to the Optimizely Feature Flags page.
- [ ] Ensure no regressions were introduced to the new comments feature.
- [ ] Ensure creator can still post comments to projects and updates via `ProjectActivitiesViewController`
- [ ] Ensure no pre-existing tests were inaccurately removed.

### CODE REVIEW

File: KsApi/models/DeprecatedAuthorTests.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @testable import KsApi
 import XCTest
 
-final class DeprecatedCommentTests: XCTestCase {
+final class DeprecatedAuthorTests: XCTestCase {
```
Comment (singhhari) [inline KsApi/models/DeprecatedAuthorTests.swift:4]: Was this just incorrectly named the entire time?

**Review by singhhari — CHANGES_REQUESTED** [review]
Did a global search on the word `Deprecated` in the project navigator and a handful of files and methods are still there. You should clean that up, replace the Deprecated models with new ones (or delete if unused) and test there are no regressions.

File: KsApi/models/DeprecatedAuthorTests.swift (hunk 1 above)
Comment (msadoon) [reply KsApi/models/DeprecatedAuthorTests.swift:4]: Haha yea it was, must not have been a member of any framework, otherwise we would have noticed.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Views/Controllers/CommentsViewControllerTests.swift:180:1 | warning: (indent) Indent code in accordance with the scope level. |
| Kickstarter-iOS/Views/Controllers/OptimizelyFeatureFlagToolsViewControllerTests.swift:9:1 | warning: (indent) Indent code in accordance with the scope level. |
| KsApi/MockService.swift:234:1 | warning: (unusedArguments) Mark unused function arguments with _. |
| KsApi/MockService.swift:235:1 | warning: (unusedArguments) Mark unused function arguments with _. |
| KsApi/Service.swift:176:1 | warning: (blankLinesBetweenScopes) Insert blank line before class, struct, enum, extension, protocol or function declarations. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by singhhari — APPROVED** [review]
lgtm

**Pull request merged**